src/rust/engine/options/src/lib.rs

```diff
@@ -199,21 +196,23 @@ pub enum Source {
 #[derive(Debug)]
 pub struct OptionValue<T> {
     pub derivation: Option<Vec<(Source, T)>>,
-    // Scalar options are always set from a single source, so we provide that
-    // here, as it can be useful in user-facing messages.
     pub source: Source,
     pub value: T,
 }
 
 #[derive(Debug)]
 pub struct ListOptionValue<T> {
     pub derivation: Option<Vec<(Source, Vec<ListEdit<T>>)>>,
+    // The highest-priority source that provided edits for this value.
```

Comment:
Not directly needed for fromfile, but will be needed for replacing the Python parser, and it was easy to sneak in here...